### PR TITLE
Use React-Helmet to Change Titles

### PR DESCRIPTION
**Pull request by wgranger**
Supercedes #3911 

Describe your changes.
Per suggestion by @srallen, this PR uses a third-party library to change doc titles. This was initially done to remove the titleMixin. 

Aside from removing the titleMixin and using react-helmet on components it touches, I've also used react-helmet on some other components to have some sort of standardized doc titles across the site.

https://use-react-helmet.pfe-preview.zooniverse.org/

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

**Comment by rogerhutchings** [comment]
Soooooo should all these title strings actually be in counterpart for translation?

_ducks_

**Comment by wgranger** [comment]
@rogerhutchings Yes... Probably so. I can make that change pretty quickly.

File: app/pages/settings/index.cjsx [hunk 1]
```diff
@@ -20,6 +21,7 @@ UserSettingsPage = React.createClass
 
   render: ->
     <section className="user-profile-content">
+      <Helmet title="#{counterpart 'userSettingsPage.title'} » #{@props.user.display_name}" />
       <div className="secondary-page settings-page">
         <h2><Translate content="userSettingsPage.header" /></h2>
```
Comment (srallen) [inline app/pages/settings/index.cjsx]: Changing the counterpart object broke this header: 
screen shot 2017-07-24 at 11 57 49 am

File: app/talk/inbox.cjsx [hunk 2]
```diff
@@ -14,17 +16,21 @@ PAGE_SIZE = talkConfig.inboxPageSize
 
 promptToSignIn = -> alert (resolve) -> <SignInPrompt onChoose={resolve} />
 
+counterpart.registerTranslations 'en',
+  messagesPage:
+    title: 'Messages'
```
Comment (srallen) [inline app/talk/inbox.cjsx]: This is inconsistent with the header and the url of the page. Maybe this should be `Inbox` instead?

**Comment by wgranger** [comment]
Good catch on the above. Title functions have been removed, aside from one with a bit more logic.

**Comment by eatyourgreens** [comment]
Ha! I've approved this but GitHub still says no to merging.

Thanks for moving some more hardcoded English strings into counterpart, by the way.

**Comment by eatyourgreens** [comment]
Maybe we can get an approval from everyone on the front-end team before this is merged 😉

**Comment by simoneduca** [comment]
My comment was only meant as a little nudge to help reducing the time PRs lie around. I assume you were being sarcastic @eatyourgreens, but just in case you weren't: although I think a review from everyone would be unnecessary, another pair of eyes can't hurt, can it?

**Comment by eatyourgreens** [comment]
Pointing out that you and I have both approved this today, so it's very much good to go, but github still makes us wait on @srallen, which is fine but would be a problem if she was away for some reason.

**Comment by simoneduca** [comment]
Gotcha, apologies for the misunderstanding. There is the [dismiss review option](https://github.com/blog/2265-dismissing-reviews-on-pull-requests), but it feels a bit rude to bypass somebody.

**Comment by eatyourgreens** [comment]
That looks really useful if someone is off sick, and a change has to be merged urgently. I agree, rude otherwise to bypass someone like that.

**Comment by srallen** [comment]
@wgranger could you update the mixin issue that this one is done? Thanks!

**Comment by wgranger** [comment]
done

**Pull request merged**